integration-testing/src/tools/fixture.py

```diff
@@ -19,7 +19,7 @@ def make_wrapper(fn, fixtures):
     namespace = {"fn":fn, "fixtures":fixtures}
 
     wrapper_code = f"""
-def wrapper(request, {parameter_list}):
+def {fn.__name__}(request, {parameter_list}):
```

Comment:
Is this runtime code generation?  This is usually a bad idea.  Why do we need this?

Reply:
I needed to parametrize the tests by fixtures and I havent' found a different way to say that a testX runs with all the combination of fixtures. 

For example I want a test to run with a network fixture, a contract generation fixture and a deployment fixture

The network fixture ca be complete, star or line shaped
The contract deployment fixture can be big, small or many comms.
The deployment fixture can be fast deployment, slow or bursts.

So I want to run my test, with the same logic in all the 27 combinations (i.e. the cartezian product of the above). 

With this decorator I would say something like this

```python
@parametrized(
    network=[complete, star, line], 
    contract_factory=[small, big, many_comms], 
    deployer=[fast, slow, bursts])
def test_fancy_stuff(network, contract_factory, deployer):
  #...
```
I wanted to use this to run test_casper_propose_and_deploy whebn the network is either  complete or star shaped network and the blocks could be valid or invalid. So that's 4 tests with an identical body. 

In the end my understanding is that we won't test the invalid blocks in the integration tests but the tool is already implemented and working well afaik so I think it's worth keeping it.